Utilities/O2device/include/O2device/O2device.h

```diff
@@ -0,0 +1,67 @@
+/********************************************************************************
+ *    Copyright (C) 2014 Frankfurt Institute for Advanced Studies               *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *         GNU Lesser General Public Licence version 3 (LGPL) version 3,        *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+/**
+ * O2device.h
+ *
+ * @since 2014-12-10
+ * @author M. Krzewicki <mkrzewic@cern.ch>
+ */
+
+#ifndef O2DEVICE_H_
```

Comment:
Since all the access helpers in FairMQdevice are just class methods which talk to the class internal state (e.g. fTransportFactory) it would be hard to implement the interface cleanly in a different way; it would also clash semantically (or at least stylistically) with the other FairMQdevice methods. But it would be relatively easy to introduce an "expert" interface for use outside of an FairMQdevice (or, in fact, O2device) - it would just be a bit awkward if we are to use fairmq.